### PR TITLE
Allowing images display in Bokeh

### PR DESCRIPTION
**Pull request by laurent-dinh**
I modified the MNIST example so that the functionality can be tested.

### CODE REVIEW

**Comment by dwf** [comment]
Docstring build failures:

```
blocks/extensions/plot.py:129 in public method `call`:
        D400: First line should end with '.', not 's'
blocks/extensions/plot.py:205 in public class `DisplayImage`:
        D209: Put multi-line docstring closing quotes on separate line preceeded by a blank line
blocks/extensions/plot.py:205 in public class `DisplayImage`:
        D202: No blank lines allowed *after* class docstring, found 1
blocks/extensions/display.py:76 in public function `reshape_image`:
        D202: No blank lines allowed *after* function docstring, found 1
blocks/extensions/display.py:173 in public class `IndexableImageDatasetDisplay`:
        D202: No blank lines allowed *after* class docstring, found 1
blocks/extensions/display.py:206 in public class `WeightDisplay`:
        D202: No blank lines allowed *after* class docstring, found 1
```

**Comment by dwf** [comment]
And your tests are failing with 

```
from blocks.extensions.plot import Plot
ImportError: cannot import name Plot
```

Sooooo that's interesting.

**Comment by vdumoulin** [comment]
@dwf Looks like `Plot` was renamed `PlotManager`.

File: blocks/extensions/plot.py [hunk 1]
```diff
@@ -139,3 +110,157 @@ def __setstate__(self, state):
...
+            fig.renderers = []
+
+            fig.image_rgba(image=[rgba_image],
+                           x=[0], y=[0],
+                           dw=[min(img_w * 1. / img_h, 1)],
+                           dh=[min(img_h * 1. / img_w, 1)])
+
+
+def format_image_to_rgba(image_0255):
+    # norm_image = image / max(abs(image).max(), 1)
```
Comment (vdumoulin) [inline blocks/extensions/plot.py]: The commented code should be removed.

**Comment by laurent-dinh** [comment]
Yes, I'll need to tie loose ends.

**Comment by rizar** [comment]
Do I understand that the application area for this would be generative modelling of images?

**Comment by vdumoulin** [comment]
I think it's more general than that. Basically anything that has to do with visualization, like looking at weights, for instance.

**Comment by bartvm** [comment]
As I mentioned to @laurent-dinh this afternoon: I think this is a nice feature, but I think advanced plotting (and maybe plotting in its entirety) would be a good candidated for `blocks-contrib`. I'll finally try and find some time to set that up!

**Comment by bartvm** [comment]
`blocks-extras` finally exists, so closing this PR here

**Pull request closed without merging**